File: Backend/WeatherService/WeatherService/src/Presentation/Models/BaseResponse.cs
namespace WeatherService.src.Presentation.Models
{
    public class BaseResponse<TType> : BaseResponse_Error
    {
        public int? ResultCode { get; set; }
        public List<string> Messages { get; set; } = new List<string>();
        public int TotalPages { get; set; }
        public int TotalRecords { get; set; }
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public TType Content { get; set; }
    }

    public class BaseResponse_Error
    {
        private readonly string defaultMessage = "No se pudo completar la operación.";
        public string ErrorCode { get; set; }
        public string ErrorMessageUser { get; set; }
        public string ErrorMessageSystem { get; set; }
        public string ErrorMessageUserDefault => defaultMessage;
    }
}

File: Frontend/React-Weather-App/src/models/BaseResponse.ts
export class BaseResponseError {
    errorCode?: string;
    errorMessageUser?: string;
    errorMessageSystem?: string;
    constructor(init?: Partial<BaseResponseError>) {
        Object.assign(this, init);
    }
}

export class BaseResponse<TType> extends BaseResponseError {
    resultCode?: number;
    messages: string[] = [];
    totalPages: number = 0;
    totalRecords: number = 0;
    currentPage: number = 0;
    pageSize: number = 0;
    content?: TType;

    constructor(init?: Partial<BaseResponse<TType>>) {
        super(init);
        if (init) {
            this.resultCode = init.resultCode;
            this.messages = init.messages || [];
            this.totalPages = init.totalPages || 0;
            this.totalRecords = init.totalRecords || 0;
            this.currentPage = init.currentPage || 0;
            this.pageSize = init.pageSize || 0;
            this.content = init.content;
        }
    }
}